parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -30,79 +30,33 @@ pub use weights::WeightInfo;
 	Encode, Decode, CloneNoBound, PartialEqNoBound, RuntimeDebugNoBound, MaxEncodedLen, TypeInfo,
 )]
 #[scale_info(skip_type_params(M))]
-#[codec(mel_bound(AccountId: MaxEncodedLen))]
-pub struct EnqueuedMessage<AccountId, M: Get<u32>>
+#[codec(mel_bound(SourceId: MaxEncodedLen))]
+pub struct Message<SourceId, M: Get<u32>>
 where
-	AccountId: Encode + Decode + Clone + PartialEq + Debug + MaxEncodedLen + TypeInfo,
+	SourceId: Parameter + Member + MaxEncodedLen,
 {
-	account: AccountId,
-	message: Message<M>,
-}
-
-#[derive(
-	Encode, Decode, CloneNoBound, PartialEqNoBound, RuntimeDebugNoBound, MaxEncodedLen, TypeInfo,
-)]
-#[scale_info(skip_type_params(M))]
-#[codec(mel_bound())]
-pub struct Message<M: Get<u32>> {
-	/// Unique message ID
+	/// ID of source parachain
+	source_id: SourceId,
+	/// Unique nonce to prevent replaying messages
 	#[codec(compact)]
```

Comment:
The nonce shouldnt be compact encoded, as that has as that it is quite expensive to parse on the Solidity side.

Reply:
Agreed, though I don't think we send any SCALE-encoded values to the Solidity side. We use SCALE for the event emitted, but everything related to the proof is ethabi-encoded. We also didn't have any special parsing for it on the Solidity side before, just `bundle.nonce` from calldata

Reply:
oh right, my mistake, that's very true